synapse/lib/syntax.py

```diff
@@ -1458,7 +1505,30 @@ def quoted(self):
 
             text += c
 
-        self._raiseSytaxError('unexpected end of query text')
+        self._raiseSyntaxError('unexpected end of query text')
+
+    def singlequoted(self):
```

Comment:
this is missing the `\\` escape logic from `quoted()` and is used interchangbly. It also has a nearly the same purpose. Can we just make the quote character for `quoted()` an arg that defaults to `"` which we can override by providing `'` when needed?  That way we don't have two different quoted/escaping rules possible.